src/loading.jl

```diff
@@ -0,0 +1,86 @@
+"""
+    loadleaf!(dst, src, err)
+
+Copy `src` to `dst` or throw `err` when their sizes are mismatched.
+By default, use `copyto!` when `dst` and `src` are arrays.
+When only `dst` is an array, set every element to `src`.
+Otherwise, just return `dst`.
+"""
+loadleaf!(dst, src, err) = dst
+function loadleaf!(dst::AbstractArray, src, err)
+  dst .= src
+  return dst
+end
+function loadleaf!(dst::AbstractArray, src::AbstractArray, err)
+  (size(dst) == size(src)) || throw(err)
+  copyto!(dst, src)
+end
+
+"""
+    loadmodel!(dst, src)
+
+Copy all the parameters (trainable and non-trainable) from `src` to `dst`.
+
+`loadmodel!` recursively walks the [`Functors.children`](@ref) of `dst` and `src`
+calling `loadleaf!` on any pair of children where [`Functors.isleaf`](@ref) is true.
+It throws an error whenever:
+- `dst` and `src` do not share the same fields (at any level)
+- the sizes of leaf nodes are mismatched between `dst` and `src`
+
+```julia
+julia> using Flux: loadmodel!
+
+julia> dst = Chain(Dense(Flux.ones32(2, 5)), Dense(2 => 1))
+Chain(
+  Dense(5 => 2),                        # 12 parameters
+  Dense(2 => 1),                        # 3 parameters
+)                   # Total: 4 arrays, 15 parameters, 316 bytes.
+
+julia> src = Chain(Dense(5 => 2), Dense(2 => 1));
+
+julia> all(isone, dst[1].weight)
+true
+
+julia> dst = loadmodel!(dst, src)
+Chain(
+  Dense(5 => 2),                        # 12 parameters
+  Dense(2 => 1),                        # 3 parameters
+)                   # Total: 4 arrays, 15 parameters, 316 bytes.
+
+julia> all(isone, dst[1].weight)
+false
```

Comment:
Maybe less cryptic? Or just omit?
```suggestion
julia> dst[1].weight ≈ ones(2, 5)
false
```